### PR TITLE
Wire internal KVS to TargetedSweeper.

### PR DESCRIPTION
**Pull request by jkozlowski**
## General
**Before this PR**:
TargetedSweeper still gets it's KVS through TransactionManagers#getKeyValueService.

**After this PR**:

==COMMIT_MSG==
Wire internal KVS to TargetedSweeper.
==COMMIT_MSG==

**Priority**:

**Concerns / possible downsides (what feedback would you like?)**:
* This is obviously still broken, the deletes from user tables should actually be going through TransactionKeyValueService, but I want to punt this for later as it does not affect correctness immediately.
* The consequences are that when running with internal migration service, eventually sweep will be doing it's deletes to wrong KVSes for particular tables, effectively removing sweep entirely and tanking performance. We will fix that eventually.

**Is documentation needed?**:

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:

**Does this PR need a schema migration?**

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:

**What was existing testing like? What have you done to improve it?**:

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:

## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:

**Has the safety of all log arguments been decided correctly?**:

**Will this change significantly affect our spending on metrics or logs?**:

**How would I tell that this PR does not work in production? (monitors, etc.)**:

**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:

**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:

**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:

**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:

## Development Process
**Where should we start reviewing?**:

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java [hunk 1]
```diff
@@ -80,7 +82,8 @@ private TargetedSweeper(
             Supplier<TargetedSweepRuntimeConfig> runtime,
             TargetedSweepInstallConfig install,
             List<Follower> followers,
-            AbandonedTransactionConsumer abandonedTransactionConsumer) {
+            AbandonedTransactionConsumer abandonedTransactionConsumer,
+            KeyValueService internalKvs) {
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java]: Can we just name this `kvs` for now? I think large internal product still goes through this.

Or maybe we should send out a note on internal dev lists saying what we're doing and why for them it is okay to be passing their normal KVSes as an `internalKvs`.

**Review by jeremyk-91 — COMMENTED** [review]
makes sense, it's kind of super weird that there is a public initializeWithoutRunning() that takes the KVS. One note on naming/comms

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java (hunk 1 above)
Comment (jkozlowski) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java]: cool, I'll rename

Comment (jkozlowski) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java]: Done

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java [hunk 2]
```diff
@@ -113,24 +117,25 @@ public static TargetedSweeper createUninitialized(
             Supplier<TargetedSweepRuntimeConfig> runtime,
             TargetedSweepInstallConfig install,
             List<Follower> followers,
-            AbandonedTransactionConsumer abandonedTransactionConsumer) {
-        return new TargetedSweeper(metrics, runtime, install, followers, abandonedTransactionConsumer);
+            AbandonedTransactionConsumer abandonedTransactionConsumer,
+            KeyValueService internalKvs) {
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java]: did we need to change this one too?

Comment (jkozlowski) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java]: doh

**Review by jeremyk-91 — APPROVED** [review]
👍

**Pull request merged**